versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/BinaryKey.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.adapter;
+
+import org.immutables.value.Value;
+
+/** Represents the composite object that represent the primary key for a binary object in Nessie. */
+@Value.Immutable
+public interface BinaryKey {
+
+  /** Content ID to which the binary object identified by this key belongs to. */
+  ContentId getContentId();
+
+  /** The object type, for example {@code snapshot} or {@code schema}. */
+  String getObjectType();
+
+  /** The object ID, for example the snapshot-ID or schema-ID. */
+  String getObjectId();
```

Comment:
Do we expect "object ID" to have weak uniqueness?.. and use "type" as part of the key to enhance its uniqueness?

Reply:
We can't really enforce it, yes. It's what we get from the "outside system" i.e. Iceberg.
There will be some checks in higher level code that prevents reusing snapshot-IDs or schema-IDs though.